pandas/tests/indexes/period/test_period_range.py

```diff
@@ -20,7 +22,7 @@ def test_required_arguments(self):
         with pytest.raises(ValueError, match=msg):
             period_range("2011-1-1", "2012-1-1", "B")
 
-    @pytest.mark.parametrize("freq", ["D", "W", "M", "Q", "A"])
+    @pytest.mark.parametrize("freq", ["D", "M", "Q", "A"])
```

Comment:
The "W" here tests for functionality that is not properly implemented in Pandas. Namely creating a `Period` from a string representing a week such as `"2017-01-23/2017-01-29"`. This only works by chance and doesn't work for weeks in the 60s, 70s, 80s, 90s of any century or any year from 2360 onwards. This is because `dateutil.parser.parse` interprets the second year as hours and minutes.

This is a bit sad because the given string is Pandas' string representation of a week period and it would be nice to have bi-directionality.

Reply:
Could you address this in a separate PR just to keep this PR tightly scoped?

Reply:
I could but doing so would stall this PR until the "W" PR is merged. The two test cases I changed raise the warning introduced in this PR. How should we proceed?

Reply:
Not sure I fully understand where and how a `"W"` frequency in particular is impacted by this new warning.

Reply:
The two tests that I propose to change create a `Period`, convert it to a `str`, and then try to create a `Period` again from the string while also telling the constructor what the `freq` is. Such a functionality isn't deliberately implemented anywhere for "W", the code path goes to `dateutil.parser.parse` which accepts these "W" period strings only if the year can be interpreted as a time. I've explained this in the [comment above](https://github.com/pandas-dev/pandas/pull/47990#discussion_r939531056) and also in #48000 - there's a section on weeks.

Reply:
Sure, this appears to be an issue, but I don't understand why these "W" changes need to made in the same PR as adding the warning for Periods with timezones since they seem independent. Having them as separate PRs makes tracking issues easier in the long run.

Reply:
This only applies to "W" because it has to do with what the string representation of a week period is and how it's parsed. Some part of that week string is interpreted as a timezone. If I remember correctly, it's the hyphen and the day (`-29` in the example). `dateutil.parser.parse` reads that as a timezone offset of negative 29 hours. Converting back to a `Period` will throw the warning introduced in this PR.

Reply:
Ah okay thanks. That explanation makes sense. 

But it would be good to still keep testing `"W"` even if it works by chance. (but maybe include a comment why it works by chance)

Reply:
And ignore the warning in the test or what? The CI requires that the test suite doesn't throw warnings, no? It sounds like implementing that would make the test a little complex. I'd also have to special-case the "W", so the warning is only ignored there and not for the other frequencies.

In general, what's the point of testing for undocumented & broken functionality like this? Do we have reason to believe that library users are relying on parsing these kind of strings? To me the conversion from Period to String and back seems like a hack to write those tests quickly. Is this something we need to support going forward?

Reply:
You can use `pytest.param("W", marks=pytest.mark.filterwarnings(...))` in the `parameterize`.

Generally, we have seen all sorts of undocumented use cases and have gotten issues when those behaviors "broke". I imagine this should be supported eventually.